### PR TITLE
Feat / Create common Accordion component

### PR DESCRIPTION
**Pull request by simonauner**
## What?

This PR:
- Tweaks the common Accordion component so that it can be re-used in both FAQ section and Comparison modal
    - Adds `AccordionContext` so keep track of which items that are expanded
    - Adds a default icon that can be reused (also possible to use your own if you want to, see stories)
- Updates FAQ section to use the updated Accordion
- Updates Comparison modal to use the updated Accordion
- Updates `react-animate-height` to latest version that supports ARIA attributes and `id`

Example usage with "single" mode, where only one item is expanded at the time:
```tsx
  <Accordion mode="single">
    <AccordionItem>
      <AccordionTrigger>
        Icon?
        <AccordionTriggerIcon />
      </AccordionTrigger>
      <AccordionContent>
        This example uses the default icon
      </AccordionContent>
    </AccordionItem>
    <AccordionItem>
      <AccordionTrigger>
        Accessible?
        <AccordionTriggerIcon />
      </AccordionTrigger>
      <AccordionContent>
        This component can be navigated with keyboard and has same ARIA attributes as Radix
      </AccordionContent>
    </AccordionItem>
  </Accordion>
```

https://user-images.githubusercontent.com/1343979/171154292-1dc6688d-1c52-4dc0-bc49-e8e6cb71722e.mov

The mode can also be "multi" and then many items can be opened at once:

https://user-images.githubusercontent.com/1343979/171154472-26d1e5ec-7138-44e1-b3b8-f3758de6c26c.mov

I chose not to use [Radix](https://www.radix-ui.com/docs/primitives/components/accordion) since I couldn't figure out how to work with getting an icon marked as "open" without resorting to their custom `data-` attributes, and it feels like we would leak implementation details of Radix out into our app with that. Maybe someone else wants to take a stab at this in the future?

I've looked at which ARIA attributes Radix uses and used the same ones here, so hopefully it should be accessible. This is the generated HTML:

```html
<div>
  <div>
    <button
      class="css-pno28e-AccordionTriggerElement e1vc6s3q0"
      id="8k125"
      aria-controls="jsssw"
      aria-expanded="false"
    >
      Icon?<svg
        color="#aaaaaa"
        class="e1vc6s3q1 css-tq6o5v-IconRoot-AccordionTriggerIcon e9r5yzb0"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          fill-rule="evenodd"
          clip-rule="evenodd"
          d="M12 14.9393L4.53033 7.46967L3.46967 8.53033L12 17.0607L20.5303 8.53033L19.4697 7.46967L12 14.9393Z"
          fill="currentColor"
        ></path>
      </svg>
    </button>
    <div
      id="jsssw"
      aria-labelledby="8k125"
      class="rah-static rah-static--height-zero"
      aria-hidden="true"
      style="height: 0px; overflow: hidden"
    >
      <div style="display: none">This example uses the default icon</div>
    </div>
  </div>
  <div>
    <button
      class="css-pno28e-AccordionTriggerElement e1vc6s3q0"
      id="67mnp"
      aria-controls="7ak33"
      aria-expanded="false"
    >
      Accessible?<svg
        color="#aaaaaa"
        class="e1vc6s3q1 css-tq6o5v-IconRoot-AccordionTriggerIcon e9r5yzb0"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          fill-rule="evenodd"
          clip-rule="evenodd"
          d="M12 14.9393L4.53033 7.46967L3.46967 8.53033L12 17.0607L20.5303 8.53033L19.4697 7.46967L12 14.9393Z"
          fill="currentColor"
        ></path>
      </svg>
    </button>
    <div
      id="7ak33"
      aria-labelledby="67mnp"
      class="rah-static rah-static--height-zero"
      aria-hidden="true"
      style="height: 0px; overflow: hidden"
    >
      <div style="display: none">
        This component can be navigated with keyboard and has same ARIA
        attributes as Radix
      </div>
    </div>
  </div>
</div>

```

## Why?

* It's nice to remove the duplicate implementations of accordions
* Accessibility is always good to improve

**Ticket(s): []**

### CODE REVIEW

**Review by robinandeer — APPROVED** [review]
Looks really nice :)

Quite a lot of complexity but I guess we could add some tests around this in future PRs since it's a very generic solution ^^

File: src/client/components/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -1,36 +1,209 @@
...
+      <AccordionContext.Provider
+        value={{
+          selectedId,
+          setSelectedId,
+          mode: mode,
+        }}
+      >
+        {children}
+      </AccordionContext.Provider>
+    </div>
```
[the remark this answers is not in the document]
Comment (simonauner) [reply src/client/components/Accordion/Accordion.tsx:40]: Yeah, true. It was just to create some encapsulation in the DOM tree, but I can remove it

**Comment by simonauner** [comment]
> Looks really nice :)
> 
> Quite a lot of complexity but I guess we could add some tests around this in future PRs since it's a very generic solution ^^

You mean with all the Contexts or with the "component API"?

**Comment by robinandeer** [comment]
> > Looks really nice :)
> > Quite a lot of complexity but I guess we could add some tests around this in future PRs since it's a very generic solution ^^
> 
> You mean with all the Contexts or with the "component API"?

Yeah I guess with the Context solution. But I think it's much more important that we hide those details from the user of the component -- so it's a great solution!

**Comment by simonauner** [comment]
> Yeah I guess with the Context solution. But I think it's much more important that we hide those details from the user of the component -- so it's a great solution!

Yeah. Let's see if we can simplify it going forward! I can def add tests as well 👍

**Pull request merged**